Modules/posixmodule.c

```diff
@@ -2712,13 +2712,14 @@ static PyObject *
 os_ttyname_impl(PyObject *module, int fd)
 /*[clinic end generated code: output=c424d2e9d1cd636a input=9ff5a58b08115c55]*/
 {
-    char *ret;
+    int ret;
+    char buffer[MAXPATHLEN];
 
-    ret = ttyname(fd);
-    if (ret == NULL) {
+    ret = ttyname_r(fd, buffer, MAXPATHLEN);
+    if (ret != 0) {
         return posix_error();
```

Comment:
I think you need to set `errno = ret` because the C library isn't required to fill `errno` with `ret`.

Reply:
Indeed only `ttyname` sets it, with `ttyname_r` you have to do it yourself. Thanks!